src/fsharp/fsc.fs

```diff
@@ -837,11 +837,20 @@ module MainModuleBuilder =
             v
 
     let productVersionToILVersionInfo (version: string) : ILVersionInfo =
-        let parseOrZero v = match System.UInt16.TryParse v with (true, i) -> i | (false, _) -> 0us
+        let parseOrZero i (v:string) =
+            let v =
+                // When i = 3 then this is the 4th part of the version.  The last part of the version can be trailed by any characters so we trim them off
+                if i <> 3 then v
+                else
+                    v |> Seq.fold(fun (finished, v) c -> match finished with
+                                                         | false when Char.IsDigit(c) -> false, v + c.ToString()
+                                                         | _ -> true, v) (false, "") |> snd
+            match System.UInt16.TryParse v with
+            | (true, i) -> i
+            | (false, _) -> 0us
```

Comment:
Thoughts on this formatting?

```fsharp
let parseOrZero i (v:string) =
    let v =
        // When i = 3 then this is the 4th part of the version.  The last part of the version can be trailed by any characters so we trim them off
        if i <> 3 then
            v
        else
            ((false, ""), v)
            ||> Seq.fold(fun (finished, v) c ->
                if finished && Char.IsDigit(c) then
                    false, v + c.ToString()
                else
                    true, v)
            |> snd
    match System.UInt16.TryParse v with
    | true, i -> i
    | false, _ -> 0us
```

Reply:
I shall make the change, since you went to the trouble of providing the alternate.  But I'm not convinced it's either better or worse, just different.